edb/edgeql/tracer.py

```diff
@@ -310,20 +310,20 @@ def get_ref_name_starstwith(
         self, ref: qlast.ObjectRef
     ) -> Set[sn.QualName]:
         refs = set()
-        prefixes = []
+        prefixes = set()
 
         if ref.module:
             # replace the module alias with the real name
             module = self.modaliases.get(ref.module, ref.module)
-            prefixes.append(f'{module}::{ref.name}')
+            prefixes.add(f'{module}::{ref.name}')
         else:
-            prefixes.append(f'{self.module}::{ref.name}')
-            prefixes.append(f'std::{ref.name}')
+            prefixes.add(f'{self.module}::{ref.name}')
+            prefixes.add(f'std::{ref.name}')
 
         for objname in self.objects.keys():
-            for prefix in prefixes:
-                if str(objname).startswith(prefix):
-                    refs.add(objname)
+            short_name = str(objname).split('@@', 1)[0]
```

Comment:
Don't we have a helper for this?

Reply:
No, this is homegrown mangling in SDL tracer, not the schema mangling.